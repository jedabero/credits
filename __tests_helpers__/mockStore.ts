// eslint-disable-next-line import/no-extraneous-dependencies
import configureMockStore from 'redux-mock-store';
import { RootState } from '../src/store';

export default configureMockStore<RootState>();
