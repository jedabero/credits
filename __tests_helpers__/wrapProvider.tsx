import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { Store } from '../src/store';

const wrapProvider = (component: ReactElement, store: Store) => (
  <Provider store={store}>{component}</Provider>
);

export default wrapProvider;
