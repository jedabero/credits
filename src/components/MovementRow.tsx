import { useCallback } from 'react';
import { EntityId } from '@reduxjs/toolkit';

import { selectMovementById, removeMovement } from '../reducers/movements';
import { useAppSelector, useAppDispatch } from '../hooks';

type MovementRowProps = {
  movementId: EntityId,
  onEdit: React.Dispatch<React.SetStateAction<EntityId | undefined>>,
  onView: React.Dispatch<React.SetStateAction<EntityId | undefined>>,
};

function MovementRow({ movementId, onEdit, onView }: MovementRowProps) {
  const dispatch = useAppDispatch();
  const handleDelete = useCallback(
    () => dispatch(removeMovement(movementId)),
    [movementId, dispatch],
  );
  const handleEdit = useCallback(() => onEdit(movementId), [movementId, onEdit]);
  const handleView = useCallback(() => onView(movementId), [movementId, onView]);
  const movement = useAppSelector((state) => selectMovementById(state, movementId));
  if (!movement) return null;
  return (
    <div className="movementrow">
      <div>{(`${movementId}`).substring(0, 13)}</div>
      <div>{movement.authNumber}</div>
      <div>{movement.date}</div>
      <div>{movement.description}</div>
      <div>{movement.value.toLocaleString('en-419', { style: 'currency', currency: 'USD' })}</div>
      <div>{movement.monthlyRate}</div>
      <div>{movement.yearlyRate}</div>
      <div>{movement.quotas}</div>
      <div>
        <button type="button" onClick={handleView}>View</button>
        <button type="button" onClick={handleEdit}>Edit</button>
        <button type="button" onClick={handleDelete}>Delete</button>
      </div>
    </div>
  );
}

export default MovementRow;
