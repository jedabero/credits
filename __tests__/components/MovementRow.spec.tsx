/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { create } from 'react-test-renderer';
import MovementRow from '../../src/components/MovementRow';
import wrapProvider from '../../__tests_helpers__/wrapProvider';
import mockStore from '../../__tests_helpers__/mockStore';

describe('MovementRow test', () => {
  test('should render', () => {
    const store = mockStore({
      _persist: { version: 0, rehydrated: true },
      creditCards: {
        entities: {},
        ids: [],
      },
      movements: {
        entities: {},
        ids: [],
      },
    });
    const tree = create(
      wrapProvider(
        <MovementRow movementId="SOME ID" onEdit={() => {}} onView={() => {}} />,
        store,
      ),
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
