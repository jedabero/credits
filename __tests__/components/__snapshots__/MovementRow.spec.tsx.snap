// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovementRow test should render 1`] = `null`;
